### PR TITLE
NESocket - Use std::string instead of String

### PR DESCRIPTION
**Pull request by biwiki**
=== NOT READY TO MERGE ===
Converting String to std::string in the areg base can be done incrementally like this.
Still not compilable, but it is quite promising that we can move to std::string step by step.

### CODE REVIEW

File: framework/areg/base/private/SocketClient.cpp [hunk 1]
```diff
@@ -37,10 +37,10 @@ bool SocketClient::createSocket( void )
 
     if ( mAddress.isValid() )
     {
-    	SOCKETHANDLE hSocket = NESocket::clientSocketConnect(static_cast<const char *>(mAddress.getHostAddress()), mAddress.getHostPort());
+        SOCKETHANDLE hSocket = NESocket::clientSocketConnect(mAddress.getHostAddress().c_str(), mAddress.getHostPort());
```
Comment (aregtech) [inline framework/areg/base/private/SocketClient.cpp]: This is OK. But when we complete converting to `std::string`, we'll need to convert `const char *` in function parameters into `std::string_view` or `const std::string &`. 
At the moment it is fine and I leave like it is.

**Comment by aregtech** [comment]
@biwiki, thank you for changes.
The MS Build fails. The line 227 in NESocket.cpp file --> please change `.isEmpty()` ==> `.empty()`.
It is MS VS specific code and do not see when compile with GCC.

**Review by aregtech — CHANGES_REQUESTED** [review]
Please check MS Build.
File NESocket.cpp, line 227 is MS VS specific code --> `isEmpty()` method does not exist.

**Comment by biwiki** [comment]
@aregtech Yes, this PR breaks the build, thus it is not ready to merge.
In fact, this is a tiny demonstration for the approach that we can use to convert String to std::string.

File: framework/areg/base/private/SocketClient.cpp (hunk 1 above)
Comment (biwiki) [reply framework/areg/base/private/SocketClient.cpp]: ACKed

**Comment by biwiki** [comment]
@aregtech please take another look at this PR, thanks!

**Review by aregtech — APPROVED** [review]
Thank you!

**Pull request merged**